.github/workflows/gh-ci.yaml

```diff
@@ -0,0 +1,279 @@
+name: mda_gh_ci
+on:
+  push:
+    branches:
+      - develop
+      - master
+  pull_request:
+    branches:
+      - develop
+      - master
+
+defaults:
+  run:
+    shell: bash -l {0}
+
+env:
+  MDA_CONDA_MIN_DEPS: "pip pytest mmtf-python biopython networkx cython matplotlib scipy griddataformats hypothesis gsd codecov"
+  MDA_CONDA_EXTRA_DEPS: "seaborn>=0.7.0 clustalw=2.1 netcdf4 scikit-learn joblib>=0.12 chemfiles tqdm>=4.43.0 tidynamics>=1.0.0 rdkit>=2020.03.1 h5py==2.10.0"
+  MDA_PIP_MIN_DEPS: 'coveralls coverage<5 pytest-cov pytest-xdist'
+  MDA_PIP_EXTRA_DEPS: 'duecredit parmed'
+
+jobs:
+  main_tests:
+    if: "github.repository == 'MDAnalysis/mdanalysis'"
+    runs-on: ${{ matrix.os }}
+    strategy:
+        fail-fast: false
+        matrix:
+          os: [ubuntu-latest, ]
+          python-version: [3.6, 3.7, 3.8]
+          run_type: [FULL, ]
+          install_hole: [true, ]
+          codecov: [true, ]
+          include:
+            - name: macOS
+              os: macOS-latest
+              python-version: 3.6
+              run_type: FULL
+              install_hole: true
+              codecov: true
+            - name: minimal-ubuntu
+              os: ubuntu-latest
+              python-version: 3.6
+              run_type: MIN
+              install_hole: false
+              codecov: true
+            - name: numpy_min
+              os: ubuntu-latest
+              python-version: 3.6
+              run_type: FULL
+              install_hole: false
+              codecov: false
+              numpy: numpy=1.16.0
+            - name: asv_check
+              os: ubuntu-latest
+              python-version: 3.6
+              run_type: FULL
+              install_hole: false
+              codecov: false
+    env:
+      CYTHON_TRACE_NOGIL: 1
+      MPLBACKEND: agg
+      GH_OS: ${{ matrix.os }}
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: setup_osx
+      if: startsWith(matrix.os, 'macOS')
+      run: |
+        # Set OS specific vars and compiler flags
+        echo "OS_NAME=osx" >> $GITHUB_ENV
+        # TODO: work out why this is necessary (from CI helpers)
+        echo "MACOSX_DEPLOYMENT_TARGET=10.9" >> $GITHUB_ENV
```

Comment:
Anyone with a MacOS device know why this is necessary? I've taken this from the astropy ci-helpers, not having it seems to cause some pip compilation errors (see: https://github.com/MDAnalysis/mdanalysis/runs/1462154629?check_suite_focus=true).

However having it also yields these types of warnings:
`ld: warning: dylib (/usr/local/miniconda/envs/test/lib/libc++.dylib) was built for newer macOS version (10.9) than being linked (10.7)`

Considering the tests pass, unless someone knows a quick answer here, I'm tempted to just leave it for now and someone with a macOS device can fix it at a later date?

Reply:
Related to https://github.com/conda/conda-build/issues/3682?

Reply:
Probably? I've read the thread twice and I honestly can't seem to capture what's being argued there 🙃 

Any ideas of potential downsides to including this?

Reply:
Is `/usr/local/miniconda/envs/test` where the GH actions install miniconda? Just curious about the path.

The error sounds almost as if `MACOSX_DEPLOYMENT_TARGET=10.9` is not correctly picked up because somewhere it is being linked against something that was build with the 10.7 SDK. Are we linking against anything outside the anaconda distribution? 

If the tests just run then I'd leave as is for the time being (and raise an issue so that it's not forgotten).

Reply:
This is on us, it's https://github.com/MDAnalysis/mdanalysis/blob/6803cf6c5aa62744099803145f3ac9398389244c/package/setup.py#L288-L292

Do we have good reason to support 10.7? It's now [10 years old.](https://en.wikipedia.org/wiki/Mac_OS_X_Lion)

Reply:
> Is /usr/local/miniconda/envs/test where the GH actions install miniconda? Just curious about the path.

Yeah, the paths are a little bit all over the place. I think the miniconda action defaults to a system-wide install, so it's `/usr/share/miniconda/envs/test` on the ubuntu builds, and `/usr/local/miniconda/envs/test` on MacOS.

As a follow up to:

> If the tests just run then I'd leave as is for the time being (and raise an issue so that it's not forgotten).

and

> Do we have good reason to support 10.7? It's now 10 years old.

I agree with @orbeckst that an issue be raised & a follow-on PR be made for this issue. Currently where we are at is that if we don't specify `MACOSX_DEPLOYMENT_TARGET` things fails, and this current workflow offers a direct translation of what we had happening on Travis CI through the astropy CI helpers.

That is to say, at this current point in time our CI is probably no worse or better than it was before we lost access to Travis.

Given that I don't have** access to MacOS device, further debugging here is probably going to have to fall to someone that does.

** I guess technically I could boot up a VMBox instance, but I'd prefer not.